.github/workflows/nightly_release.yml

```
@@ -0,0 +1,93 @@
+name: Nightly Builds
+on:
+  schedule:
+    # Runs every weekday at 1am
+    - cron: 0 1 * * 1-5
+  workflow_dispatch:
+
+
+jobs:
+  tag_nightly_release:
+    name: Tag a new nightly release
+    runs-on: ubuntu-latest
+    outputs:
+      tag_name: ${{ steps.set_tagname.outputs.tag }}
+
+    steps:
+      - name: Checkout git repository 🕝
+        uses: actions/checkout@v3
+
+      - name: Compose tag name
+        id: set_tagname
+        run: |
+          DATE=$(date +'%Y%m%d')
+
+          # Find latest rasa-oss version
+          echo "Trying to find the latest rasa-oss version..."
+          LATEST_RASA_VERSION=$(python -c "import sys; import os; sys.path.append('src/rasa/rasa'); from version import __version__; print(__version__)")
+          echo "Current RASA version: ${LATEST_RASA_VERSION}"
+
+          LATEST_NIGHTLY_VERSION=$(echo ${LATEST_RASA_VERSION} | awk -F. '{$(NF-1) = $(NF-1) + 1;} {$(NF) = 0;} 1' | sed 's/ /./g')
+
+          echo "New nightly release version: ${LATEST_NIGHTLY_VERSION}"
+
+          echo "Composing nightly build tag name..."
+          GH_TAG=${LATEST_NIGHTLY_VERSION}-dev${DATE}
```

can we follow [pep440](https://peps.python.org/pep-0440/#developmental-releases) here and use a `.`? AFAIK our tag names are exactly our python version names.
```suggestion
          GH_TAG=${LATEST_NIGHTLY_VERSION}.dev${DATE}
```

---

This is actually a very interesting rabbit hole to get into, I was wondering if semvar has any opinion on the matter and found this: https://github.com/semver/semver/issues/200

But I agree with @m-vdb , let's go with the `.`.